### ISSUE TITLE
Delete files associated with old documentation

### ISSUE THREAD

**Issue by hayleycd** [google/osv.dev#1217]
I opened #1216 to remove old documentation from the /docs folder. 

However, there are other associated files that should be removed once the redirects are live. 

These files  include but are not limited to:

1. `gcp/appengine/docs/index.html`
2. `gcp/appengine/frontend3/src/templates/faq.html`

And there are other files that may need to be edited, for example `gcp/appengine/frontend_handlers.py`

In summary:

Remove files that are no longer needed because of the doc migration, and edit files as required to reflect the new docs.

**Comment by hayleycd** [comment]
Partially addressed by #1217
